src/database/mongodb/mode/collectionstatistics.pm

```diff
@@ -115,21 +168,41 @@ sub manage_selection {
 
         my $collections = $options{custom}->list_collections(database => $database);
 
-        $self->{databases}->{$database}->{display} = $database;
+        $self->{databases}->{$database} = {
+            display => $database,
+            collections => {},
+            shards => {}
+        };
 
         foreach my $collection (sort @{$collections}) {
             my $cl_stats = $options{custom}->run_command(
                 database => $database,
-                command => $options{custom}->ordered_hash(collStats => $collection),
+                command => $options{custom}->ordered_hash(collStats => $collection)
             );
-            
+
             $self->{databases}->{$database}->{collections}->{$collection} = {
-                display => $collection,
                 storageSize => $cl_stats->{storageSize},
                 totalIndexSize => $cl_stats->{totalIndexSize},
                 count => $cl_stats->{count},
-                nindexes => $cl_stats->{nindexes},
+                nindexes => $cl_stats->{nindexes}
             };
+
+            if (defined($self->{option_results}->{add_shards}) && defined($cl_stats->{shards})) {
+                foreach my $shard_name (keys %{$cl_stats->{shards}}) {
+                    next if (defined($self->{option_results}->{filter_shard}) && $self->{option_results}->{filter_shard} ne '' 
+                        && $shard_name !~ /$self->{option_results}->{filter_shard}/);
+
+                    $self->{databases}->{$database}->{shards}->{$collection . $shard_name} = {
+                        dbName => $database,
+                        collectionName => $collection,
+                        shardName => $shard_name,
+                        storageSize => $cl_stats->{shards}->{$shard_name}->{storageSize},
+                        totalIndexSize => $cl_stats->{shards}->{$shard_name}->{totalIndexSize},
+                        count => $cl_stats->{shards}->{$shard_name}->{count},
+                        nindexes => $cl_stats->{shards}->{$shard_name}->{nindexes}
+                    };
+                }
+            }
         }
     }
 
```

Comment:
Catch error if filter_shard option return empty result should be add here 
Something like this :

```
    if (scalar(keys %{$self->{databases}->{$database}->{shards}}) <= 0) {
        $self->{output}->add_option_msg(short_msg => 'No shard found with this filter');
        $self->{output}->option_exit();
    }
```

Reply:
shards are optional.  So we don't want to have an error if we don't have it on a mongodb

Reply:
Indeed so maybe add filter condition to cach emptu result :

```
if (defined($self->{option_results}->{filter_shard}) && scalar(keys %{$self->{databases}->{$database}->{shards}}) <= 0) {
        $self->{output}->add_option_msg(short_msg => 'No shard found with this filter');
        $self->{output}->option_exit();
    }
```